packages/core/src/utils/TypeUtil.ts

```diff
@@ -200,3 +200,31 @@ export type Event =
       type: 'track'
       event: 'CLICK_GET_WALLET'
     }
+
+// -- SIWEController Types ---------------------------------------------------
+
+export interface SIWESession {
+  address: string
+  chainId: number
+}
+
+export interface CreateSIWEMessageArgs {
+  nonce: string
+  address: string
+  chainId: number
+}
+
+export interface VerifySIWEMessageArgs {
+  message: string
+  signature: string
+}
+
+// eslint-disable-next-line no-shadow
+export enum SIWEStatus {
```

Comment:
We try to avoid enums, instead you can do

```
type SIWEStatus = 'uninitialized' | 'ready' | 'loading' ...
```
This gives you better string based autocomplete without the need to import enum every time